### ISSUE TITLE
Wmisleading-indentation not firing

### ISSUE THREAD

**Issue by m-gupta** [llvm/llvm-project#57240]
This was reported by an internal user.
This code has a misleading indent after  if (foo == 0) condition but Wmisleading-indentation is not firing (https://godbolt.org/z/8GW5ET3cb)

```C
#include <stdio.h>
#include <stdlib.h>

static void inc(volatile int *v)
{
	*v++;
}

static int foobar(void)
{
	volatile int foo = 0;
	volatile int bar = 0;

	foo = rand() % 10;

	if (foo == 0) {
		inc(&bar);
	inc(&foo);
	printf("foo %d bar %d\n", foo, bar);
	}
	return foo;
}

int main()
{
	foobar();
	return 0;
}
```

**Comment by tbaederr** [comment]
The warning is for _misleading_ indentation, not wrong indentation. Your if statement even uses braces, so the warning will never do anything.

This code warns:
```c++
	if (foo == 0)
		inc(&bar);
		inc(&foo);
```

https://godbolt.org/z/7jadGdYhG

**Comment by m-gupta** [comment]
thanks for clarifying.

**Comment by sergey-senozhatsky** [comment]
> The warning is for _misleading_ indentation, not wrong indentation. Your if statement even uses braces, so the warning will never do anything.

So can `Wmisleading-indentation` be extended to cover the entire { ... } block?

**Comment by sergey-senozhatsky** [comment]
Or can there be, say, `Wwrong-indentation` to catch such cases?

**Comment by tbaederr** [comment]
I mean... everything is possible, but you should just use clang-format I guess?

**Comment by sergey-senozhatsky** [comment]
What we had was a problem in the 3rd party linux kernel code, which caused an unrecoverable kernel error (kernel panic), and it looked, schematically, as follows

```
void foo(void)
{
    local_irq_save(flags);
    ...
    if (bar) {
         buz();
         foobar();
    local_irq_restore(flags);
    }
}
```
Clang-format, *I assume*, would fix the indentation with no complaints, whilst compiler warning could alarm us and trigger our attention.
